shacl12-ui/index.html

```diff
@@ -801,8 +801,636 @@ <h3>Document Conventions</h3>
+          </p>
+        </section>
+        <section id="EnumSelectEditor">
+          <h4>shui:EnumSelectEditor</h4>
+          <p>
+            <b>Score:</b>
+            <ul>
+              <li>10 if there exists a <code>sh:in</code> constraint for the same property at the current focus node.</li>
+              <li>0 otherwise.</li>
+            </ul>
+          </p>
+          <p>
+            <b>Rendering:</b>
+            a drop-down editor for enum fields (based on the <code>sh:in</code> list, in that order).
+          </p>
+          <img src="images/editors/EnumSelectEditor.png" />
+          <pre class="example">
+ex:AustralianAddressShape-addressRegion
+	a sh:PropertyShape ;
+	sh:path schema:addressRegion ;
+	<b>sh:in ( "ACT" "NSW" "NT" "QLD" "SA" "TAS" "VIC" "WA" ) ;</b>
+	...</pre>
+        </section>
+        <section id="InstancesSelectEditor">
+          <h4>shui:InstancesSelectEditor</h4>
+          <p>
+            <b>Score:</b>
+            <ul>
+              <li><code>null</code> if there exists a <code>sh:class</code> for the property.</li>
+              <li>0 otherwise.</li>
+            </ul>
+          </p>
+          <p>
+            <b>Rendering:</b>
+            a drop-down editor for all instances of the target class (based on <code>sh:class</code> of the property).
+            Typically only used for classes that have few instances.
+          </p>
+          <pre class="example">
+ex:Person-homeCountry
+	a sh:PropertyShape ;
+	sh:path ex:homeCountry ;
+	<b>sh:class ex:Country ;
+	shui:editor shui:InstancesSelectEditor ;</b>
+	...</pre>
+        </section>
+        <section id="RichTextEditor">
+          <h4>shui:RichTextEditor</h4>
+          <p>
+            <b>Score:</b>
+            <ul>
+              <li>10 for <code>rdf:HTML</code> literals.</li>
+              <li>0 otherwise.</li>
+            </ul>
+          </p>
+          <p>
+            <b>Rendering:</b>
+            a rich text editor to enter the lexical value of a literal and a drop-down to select language.
+            The selected language is stored in the HTML <code>lang</code> attribute of the root node in the HTML DOM tree.
+          </p>
+          <img src="images/editors/RichTextEditor.png" />
+          <pre class="example">
+ex:Concept-definition
+	a sh:PropertyShape ;
+	sh:path skos:definition ;
+	<b>sh:datatype rdf:HTML ;</b>
+	...</pre>
+        </section>
+        <section id="SubClassEditor">
+          <h4>shui:SubClassEditor</h4>
+          <p>
+            <b>Score:</b>
+            <code>null</code>, i.e., this should be selected explicitly through a <code>shui:editor</code> statement.
+            However, this widget is typically only used if the property has a
+            <a href="https://datashapes.org/constraints.html#RootClassConstraintComponent"><code>dash:rootClass</code></a>
+            constraint, or (at minimum) only allows classes as values.
```

Comment:
```suggestion
            <ul>
              <li><code>null</code>, i.e., this should be selected explicitly through a <code>shui:editor</code> statement.
                  However, this widget is typically only used if the property has a
                  <a href="https://datashapes.org/constraints.html#RootClassConstraintComponent"><code>dash:rootClass</code></a>
                  constraint, or (at minimum) only allows classes as values.
               </li>
             </ul>
```
for consistency with the other scoring parts.